easybuild/tools/run.py

```diff
@@ -589,3 +589,59 @@ def parse_log_for_error(txt, regExp=None, stdout=True, msg=None):
                   (regExp, '\n'.join([x[0] for x in res])))
 
     return res
+
+
+def check_log_for_errors(logTxt, regExps):
+    """
+    Check logTxt for messages matching regExps in order and do appropriate action
+    :param logTxt: String containing the log, will be split into individual lines
+    :param regExps: List of: regular expressions (as RE or string) to error on,
+                    or tuple of regular expression and action (any of [IGNORE, WARN, ERROR])
+    """
+    global errors_found_in_log
+
+    def is_regexp_object(objToTest):
+        try:
+            objToTest.match('')
+            return True
+        except AttributeError:
+            return False
+
+    # Avoid accidentally passing a single element
+    assert isinstance(regExps, list), "regExps must be a list"
```

Comment:
Why not promote a single string value to a single-element list, so you can use `check_log_for_errors("error:")` for example?

```python
# promote single string value to list, since code below expects a list
if isinstance(regexps, string_type):
    regexps = [regexps]
```

Reply:
Because I'd expect usually more than 1 regexp. At least a few "IGNORE" in front. And on call site you can always use `check_log_for_errors(["error:"])`. This also avoids additional complexity (I was already hesitant to allow either a tuple or a string/re)

Reply:
I still think it's convenient to promote a single value to a list, we do this in several other places too, so you can just use `check_log_for_errors("error:")`

Reply:
Ok, done